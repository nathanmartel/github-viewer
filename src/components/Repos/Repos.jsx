import React from 'react';
import PropTypes from 'prop-types';
import Repo from '../Repo/Repo';


const Repos = ({ repos, user }) => {
  const repoElements = repos.map(repo => (
    repo.name && 
      <li key={repo.id}>
        <Repo name={repo.name} user={user} />
      </li>
  ));
  
  return (
    <ul>
      {repoElements}
    </ul>
  );
};

Repos.propTypes = {
  user: PropTypes.string.isRequired,
  repos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired
  })).isRequired
};

export default Repos;
