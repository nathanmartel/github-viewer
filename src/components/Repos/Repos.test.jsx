import React from 'react';
import { shallow } from 'enzyme';
import Repos from './Repos';

describe('Repos component', () => {
  it('renders Repos', () => {
    const repos = [{ 
      id: 1,
      name: 'Lorem Ipsum' 
    }, 
    { 
      id: 2,
      name: 'Dolor' 
    }];
    const wrapper = shallow(<Repos repos={repos} user={'username'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
