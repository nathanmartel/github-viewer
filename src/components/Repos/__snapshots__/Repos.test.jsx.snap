// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component renders Repos 1`] = `
<ul>
  <li
    key="1"
  >
    <Repo
      name="Lorem Ipsum"
      user="username"
    />
  </li>
  <li
    key="2"
  >
    <Repo
      name="Dolor"
      user="username"
    />
  </li>
</ul>
`;
